src/python/pants/testutil/engine/util.py

```diff
@@ -197,3 +199,10 @@ def green(self, text):
 
   def red(self, text):
     return self._safe_color(text, red)
+
+
+def mock_goal_options(goal: Type[Goal], scoped_options: Dict[str, Any]) -> Goal.Options:
+  options_container = OptionValueContainer()
+  for option, val in scoped_options.items():
+    setattr(options_container, option, RankedValue(RankedValue.HARDCODED, val))
+  return goal.Options(scope=goal.name, scoped_options=options_container)
```

Comment:
So: basically none of this is specific to goals. This mocking would work fine for any `Optionable`.

I think that this could be:
```
def mock_options(optionable: Type[Optionable], scoped_options: Dict[str, Any]) -> Optionable:
  options_container = OptionValueContainer()
  for option, val in scoped_options.items():
    setattr(options_container, option, RankedValue(RankedValue.HARDCODED, val))
  return optionable(scope=optionable.options_scope, scoped_options=options_container)
```

Reply:
But, also... this isn't mocking. This is an actual instance.